icicle/appUtils/msm/msm.cu

```diff
@@ -180,7 +181,8 @@ void bucket_method_msm(unsigned bitsize, unsigned c, S *h_scalars, A *h_points,
   //split scalars into digits
   NUM_THREADS = 1 << 10;
   NUM_BLOCKS = (size * (nof_bms+1) + NUM_THREADS - 1) / NUM_THREADS;
-  split_scalars_kernel<<<NUM_BLOCKS, NUM_THREADS>>>(bucket_indices + size, point_indices + size, scalars, size, msm_log_size, nof_bms, bm_bitsize, c); //+size - leaving the first bm free for the out of place sort later
+  split_scalars_kernel<<<NUM_BLOCKS, NUM_THREADS>>>(bucket_indices + size, point_indices + size, on_device ? scalars : d_scalars, size, msm_log_size, 
```

Comment:
instead of using "?" every time just use the same pointer. decide only at the beginning to copy or not. It's also confusing because d_scalars and scalars are actually both on device..